File: java/Samples/Devices/CS/DevicesSampleCS/DeviceManagerForm.cs
using System;
using System.Windows.Forms;
using Neurotec.Devices;

namespace Neurotec.Samples
{
	public partial class DeviceManagerForm : Form
	{
		#region Public constructor

		public DeviceManagerForm()
		{
			InitializeComponent();

			anyCheckBox.Tag = NDeviceType.Any;
			captureDeviceCheckBox.Tag = NDeviceType.CaptureDevice;
			microphoneCheckBox.Tag = NDeviceType.Microphone;
			cameraCheckBox.Tag = NDeviceType.Camera;
			biometricDeviceCheckBox.Tag = NDeviceType.BiometricDevice;
			fScannerCheckBox.Tag = NDeviceType.FScanner;
			fingerScannerCheckBox.Tag = NDeviceType.FingerScanner;
			palmScannerCheckBox.Tag = NDeviceType.PalmScanner;
			irisScannerCheckBox.Tag = NDeviceType.IrisScanner;
			DeviceTypes = NDeviceType.Any;
		}

		#endregion

		#region Public properties

		public NDeviceType DeviceTypes
		{
			get
			{
				var value = NDeviceType.None;
				foreach (CheckBox checkBox in deviceTypesGroupBox.Controls)
				{
					if (checkBox.Checked) value |= (NDeviceType)checkBox.Tag;
				}
				return value;
			}
			set
			{
				foreach (CheckBox checkBox in deviceTypesGroupBox.Controls)
				{
					checkBox.Checked = (value & (NDeviceType)checkBox.Tag) != 0;
				}
			}
		}

		public bool AutoPlug
		{
			get
			{
				return cbAutoplug.Checked;
			}
			set
			{
				cbAutoplug.Checked = value;
			}
		}

		#endregion

		#region Private form events

		private void deviceTypeCheckBox_Click(object sender, EventArgs e)
		{
			var checkBox = (CheckBox)sender;
			var deviceType = (NDeviceType)checkBox.Tag;
			DeviceTypes = checkBox.CheckState == CheckState.Unchecked ? DeviceTypes | deviceType : DeviceTypes & ~deviceType;
		}

		private void btnAccept_Click(object sender, EventArgs e)
		{
			if (DeviceTypes == NDeviceType.None)
			{
				DeviceTypes = NDeviceType.Any;
			}
		}

		#endregion
	}
}

File: java/Samples/Devices/VB/DevicesSampleVB/DeviceManagerForm.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Windows.Forms
Imports Neurotec.Devices

Partial Public Class DeviceManagerForm
	Inherits Form
#Region "Public constructor"

	Public Sub New()
		InitializeComponent()

		anyCheckBox.Tag = NDeviceType.Any
		captureDeviceCheckBox.Tag = NDeviceType.CaptureDevice
		microphoneCheckBox.Tag = NDeviceType.Microphone
		cameraCheckBox.Tag = NDeviceType.Camera
		biometricDeviceCheckBox.Tag = NDeviceType.BiometricDevice
		fScannerCheckBox.Tag = NDeviceType.FScanner
		fingerScannerCheckBox.Tag = NDeviceType.FingerScanner
		palmScannerCheckBox.Tag = NDeviceType.PalmScanner
		irisScannerCheckBox.Tag = NDeviceType.IrisScanner
		DeviceTypes = NDeviceType.Any
	End Sub

#End Region

#Region "Public properties"

	Public Property DeviceTypes() As NDeviceType
		Get
			Dim value As NDeviceType = NDeviceType.None
			For Each checkBox As CheckBox In deviceTypesGroupBox.Controls
				If checkBox.Checked Then
					value = value Or CType(checkBox.Tag, NDeviceType)
				End If
			Next checkBox
			Return value
		End Get
		Set(ByVal value As NDeviceType)
			For Each checkBox As CheckBox In deviceTypesGroupBox.Controls
				checkBox.Checked = (value And CType(checkBox.Tag, NDeviceType)) <> 0
			Next checkBox
		End Set
	End Property

	Public Property AutoPlug() As Boolean
		Get
			Return cbAutoplug.Checked
		End Get
		Set(ByVal value As Boolean)
			cbAutoplug.Checked = value
		End Set
	End Property

#End Region

#Region "Private form events"

	Private Sub deviceTypeCheckBox_Click(ByVal sender As Object, ByVal e As EventArgs) Handles captureDeviceCheckBox.Click, microphoneCheckBox.Click, anyCheckBox.Click, fScannerCheckBox.Click, cameraCheckBox.Click, irisScannerCheckBox.Click, palmScannerCheckBox.Click, fingerScannerCheckBox.Click, biometricDeviceCheckBox.Click
		Dim checkBox As CheckBox = CType(sender, CheckBox)
		Dim deviceType As NDeviceType = CType(checkBox.Tag, NDeviceType)
		DeviceTypes = If(checkBox.CheckState = CheckState.Unchecked, DeviceTypes Or deviceType, DeviceTypes And (Not deviceType))
	End Sub

	Private Sub btnAccept_Click(ByVal sender As Object, ByVal e As EventArgs) Handles btnAccept.Click
		If DeviceTypes = NDeviceType.None Then
			DeviceTypes = NDeviceType.Any
		End If
	End Sub

#End Region
End Class
